src/platform/DXJoystick.cxx

```diff
@@ -255,33 +255,29 @@ bool          DXJoystick::joystick() const
     return (device != NULL);
 }
 
-void          DXJoystick::getJoy(int& x, int& y)
+void          DXJoystick::getJoy(float& x, float& y)
 {
     if (!device) return;
 
     DIJOYSTATE state = pollDevice();
 
-    if (xAxis == "X")  x = state.lX;
-    else if (xAxis == "Y")  x = state.lY;
-    else if (xAxis == "Z")  x = state.lZ;
-    else if (xAxis == "Rx") x = state.lRx;
-    else if (xAxis == "Ry") x = state.lRy;
-    else if (xAxis == "Rz") x = state.lRz;
-    else if (xAxis == "Slider 1") x = state.rglSlider[0];
-    else if (xAxis == "Slider 2") x = state.rglSlider[1];
-
-    if (yAxis == "X")  y = state.lX;
-    else if (yAxis == "Y")  y = state.lY;
-    else if (yAxis == "Z")  y = state.lZ;
-    else if (yAxis == "Rx") y = state.lRx;
-    else if (yAxis == "Ry") y = state.lRy;
-    else if (yAxis == "Rz") y = state.lRz;
-    else if (yAxis == "Slider 1") y = state.rglSlider[0];
-    else if (yAxis == "Slider 2") y = state.rglSlider[1];
-
-    // ballistics
-    x = (x * abs(x)) / 1000;
-    y = (y * abs(y)) / 1000;
+    if (xAxis == "X")  x = float(state.lX) / 1000.0f;
```

Comment:
lots of superfluous casting here (but at least they are function-style casts and not C-style casts)... the divide by float will do the type promotion, and the type of `x` will narrow as needed.

Reply:
Fixed... thanks.